### PR TITLE
🔒 Display password criteria in singnup screen

### LINKED ISSUE #7331: Include the password strength criteria in the Android UI
## 🎯 What needs to be done?
Inform the user of our password strength rules. If they try to signup with a password that doesn't match these rules, inform them even more [in red] 😃  

## 📝 Detailed explanation
Design links an UX: https://www.notion.so/toggl/UX-documentation-for-password-strenght-project-done-9779bca9938b46de8d1c6761b9f30bd3#89d2f5ae71a5444cbce72910a2e1579b

tldr:
- display the "Include 8 or more characters, with at least one uppercase letter, lowercase letter, and a number." below password field
- make sure the copy is always visible when the password field is active
- if user tries to signup with a password that doesn't match the criteria, make the copy red
- the copy turns white again, as soon as they type in a strong password (no need to actually press 'signup' again)

## 🔗 Dependencies
#7329 

## 🏆 Outcome
- The copy is displayed below the password field
- It is always visible when the password field is active
- If the user tries to signup with a password that doesn't match the rules, make the copy red
- The copy turns white again, as soon as they type in a strong password (no need to actually press 'signup' again)

## 👕 Estimated complexity
S

### PR DESCRIPTION
**Pull request by pdrj**
## What's this?
This displays password criteria in singnup screen. It also cleans up the IsValid property, which was used wrong in some places and turns out is not needed anymore.

### Relationships

Closes #7331 

## Why do we want this?
To let users know what's going on.

## How is it done?
To display the criteria, I'm reusing the 'error' label on textinput. It gives us some animations and styling will always be the same as of the error label on email field. I think it's better than adding a separate textview, but let me know what you think.

It turns out `IsValid` is not needed anymore, because the only places that still used it, used it incorrectly. For instance, in token reset screen, we wouldn't let users reset the token with a short password and that is wrong (some users may still be on short passwords). I've modified those usages to just check if password is non-empty and then removed the `IsValid` property.

## Review hints
- Try signing up with a weak password. The criteria should turn red. Then modify the password so that criteria are matched **the criteria should turn to regular text color as soon as they are matched, without actually pressing 'signup' again**. Delete some characters so that criteria are not matched again - try to signup and validate criteria turn red again. 
- Try to actually signup and validate it works.
 

## Localizations
#7332 

## :squid: Permissions
Anyone

### CODE REVIEW

File: Toggl.Droid/Activities/SignUpActivity.cs [hunk 1]
```diff
@@ -115,5 +123,31 @@ string signupButtonTitle(bool isLoading, string currentLoadingMessage)
...
+        {
+            if (isStrong)
+            {
+                setPasswordErrorLabelTextColor("");
+            }
+        }
+
+        private void setPasswordErrorLabelTextColor(string errorText)
+        {
+            if (errorText == "")
```
Comment (heytherewill) [inline Toggl.Droid/Activities/SignUpActivity.cs:137]: This is really cryptic, why not just make this method receive and errorDrawable, a color and a string?

File: Toggl.Shared/Resources.resx [hunk 2]
```diff
@@ -175,6 +175,10 @@ mobile, desktop or web</value>
         <value>Password should be at least 6 characters</value>
         <comment>Displayed when the user tries to login or sign up with an invalid password (less than 6 characters)</comment>
     </data>
+    <data name="StrongPasswordCriteria" xml:space="preserve">
+        <value>Include 8 or more characters, with at least one uppercase letter, lowercase letter, and a number.</value>
```
Comment (heytherewill) [inline Toggl.Shared/Resources.resx:179]: pt-br: Inclua 8 ou mais caracteres, com ao menos uma letra maiúscula, uma letra minúscula, e um número.

Comment (pdrj) [reply Toggl.Shared/Resources.resx:179]: Thanks, added to #7332

File: Toggl.Droid/Activities/SignUpActivity.cs [hunk 3]
```diff
@@ -67,15 +69,15 @@ protected override void InitializeBindings()
...
                 .DisposedBy(DisposeBag);
 
             ViewModel.SignUpError
-                 .Subscribe(errorLabel.Rx().TextObserver())
-                 .DisposedBy(DisposeBag);
+                .Subscribe(errorLabel.Rx().TextObserver())
+                .DisposedBy(DisposeBag);
 
             var animatedLoadingMessage = TextHelpers.AnimatedLoadingMessage();
-
```
Comment (pdrj) [inline Toggl.Droid/Activities/SignUpActivity.cs:70]: I must have pressed reformat hotkey out of habit, sorry.

File: Toggl.Droid/Activities/SignUpActivity.cs [hunk 4]
```diff
@@ -115,5 +125,32 @@ string signupButtonTitle(bool isLoading, string currentLoadingMessage)
...
+            {
+                setPasswordErrorLabelTextColor(error, errorTextColor, defaultErrorDrawable);
+            }
+        }
+
+        private void resetPasswordErrorLabelColorIfPasswordStrong(bool isStrong)
+        {
+            if (isStrong)
+            {
+                setPasswordErrorLabelTextColor("", secondaryTextColor, null);
```
Comment (heytherewill) [inline Toggl.Droid/Activities/SignUpActivity.cs:145]: Shouldn't this be `handlePasswordError("")`?

Comment (pdrj) [reply Toggl.Droid/Activities/SignUpActivity.cs:145]: Yes. Good catch - thanks.

**Pull request merged**